src/services/userService.js

```diff
@@ -0,0 +1,41 @@
+const { generateUniqueId } = require('../utils/getRandomId');
+
+let users = [];
+
+const start = () => {
+  users = [];
+};
+
+const getUsers = () => {
+  return users;
+};
+
+const getUserById = (id) => {
+  return users.find((user) => user.id === id) || null;
+};
+
+const create = (name) => {
+  const newUser = {
+    id: generateUniqueId(),
+    name,
+  };
+
+  users.push(newUser);
+
+  return newUser;
+};
+
+const remove = (id) => {
+  users = users.filter((user) => user.id !== id);
+};
+
+const update = (id, fields) => Object.assign(getUserById(id), { ...fields });
```

Comment:
The `update` function directly modifies the object returned by `getUserById`. Ensure that `getUserById` does not return `null` before attempting to update, as this will cause an error if the user is not found.